### DISCUSSION TITLE
Test suite

### DISCUSSION THREAD

**Discussion by HeetVekariya** [MDAnalysis/mdanalysis#4381] [Q&A] [answered]
- I am getting me familiarise by reading the test suite docs.
- While going through the test suite docs, i run below command which gives 4 errors:

```bash
pytest  --disable-pytest-warnings
```

## **Errors**:
1) **test_align.py**
```bash
_____________________________________ ERROR collecting MDAnalysisTests/analysis/test_align.py _____________________________________
ImportError while importing test module '/home/heet/Work/Development/mdanalysis/testsuite/MDAnalysisTests/analysis/test_align.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
/home/heet/Work/Development/mdanalysis/testsuite/MDAnalysisTests/analysis/test_align.py:27: in <module>
    from MDAnalysis.analysis.align import HAS_BIOPYTHON
E   ImportError: cannot import name 'HAS_BIOPYTHON' from 'MDAnalysis.analysis.align' (/home/heet/anaconda3/envs/mdanalysis-dev/lib/python3.10/site-packages/MDAnalysis/analysis/align.py)
```

2) **test_leaflet.py**
```bash
____________________________________ ERROR collecting MDAnalysisTests/analysis/test_leaflet.py ____________________________________
ImportError while importing test module '/home/heet/Work/Development/mdanalysis/testsuite/MDAnalysisTests/analysis/test_leaflet.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
/home/heet/Work/Development/mdanalysis/testsuite/MDAnalysisTests/analysis/test_leaflet.py:29: in <module>
    from MDAnalysis.analysis.leaflet import LeafletFinder, optimize_cutoff, HAS_NX
E   ImportError: cannot import name 'HAS_NX' from 'MDAnalysis.analysis.leaflet' (/home/heet/anaconda3/envs/mdanalysis-dev/lib/python3.10/site-packages/MDAnalysis/analysis/leaflet.py)
```

3) **test_nucleicacids.py**
```bash
_________________________________ ERROR collecting MDAnalysisTests/analysis/test_nucleicacids.py __________________________________
ImportError while importing test module '/home/heet/Work/Development/mdanalysis/testsuite/MDAnalysisTests/analysis/test_nucleicacids.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
/home/heet/Work/Development/mdanalysis/testsuite/MDAnalysisTests/analysis/test_nucleicacids.py:30: in <module>
    from MDAnalysis.analysis.nucleicacids import (NucPairDist, WatsonCrickDist,
E   ImportError: cannot import name 'MajorPairDist' from 'MDAnalysis.analysis.nucleicacids' (/home/heet/anaconda3/envs/mdanalysis-dev/lib/python3.10/site-packages/MDAnalysis/analysis/nucleicacids.py)
```

4) **test_residugroup.py**
```bash
___________________________________ ERROR collecting MDAnalysisTests/core/test_residuegroup.py ____________________________________
ImportError while importing test module '/home/heet/Work/Development/mdanalysis/testsuite/MDAnalysisTests/core/test_residuegroup.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
/home/heet/Work/Development/mdanalysis/testsuite/MDAnalysisTests/core/test_residuegroup.py:29: in <module>
    from MDAnalysis.core.topologyattrs import HAS_BIOPYTHON
E   ImportError: cannot import name 'HAS_BIOPYTHON' from 'MDAnalysis.core.topologyattrs' (/home/heet/anaconda3/envs/mdanalysis-dev/lib/python3.10/site-packages/MDAnalysis/core/topologyattrs.py)
```

<br>

## **Overall summery**:
```bash
============================================================ short test summary info =============================================================
ERROR analysis/test_align.py
ERROR analysis/test_leaflet.py
ERROR analysis/test_nucleicacids.py
ERROR core/test_residuegroup.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 4 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
==================================================== 3 skipped, 6 warnings, 4 errors in 9.43s ====================================================
```

- Does this allowed to persist or needed to be solved in another issue?

**Comment by HeetVekariya** [comment]
- @IAlibay can you give me information about what/how development environment should be initialised ?
- I am doing contributions to MDAnalysis, so am i in the development environment or not ?

**Reply by IAlibay** [reply, marked as answer]
@HeetVekariya please see here for more information on how to set up a development environment: https://userguide.mdanalysis.org/stable/contributing_code.html#creating-a-development-environment

**Reply by HeetVekariya** [reply]
- I have already performed the steps of the docs you have referred me for setting up the development environment. 
- The errors i have shown above is also observed in that environment only.

**Reply by IAlibay** [reply]
@HeetVekariya I can confirm that these tests failures are not happening locally or on our CI suite. This seems like an installation mishap.

Could you run `python -Ic "import MDAnalysisTests; print(MDAnalysisTests.__version__)"` in your development environment? If so, what value do you get?

**Reply by HeetVekariya** [reply]
Output:
```bash
(mdanalysis-dev) heet@heet:~/Work/Development/mdanalysis/testsuite/MDAnalysisTests$ python -Ic "import MDAnalysisTests; print(MDAnalysisTests.__ve
rsion__)"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/heet/anaconda3/envs/mdanalysis-dev/lib/python3.10/site-packages/MDAnalysisTests/__init__.py", line 95, in <module>
    import pytest
  File "/home/heet/anaconda3/envs/mdanalysis-dev/lib/python3.10/site-packages/pytest/__init__.py", line 5, in <module>
    from _pytest._code import ExceptionInfo
  File "/home/heet/anaconda3/envs/mdanalysis-dev/lib/python3.10/site-packages/_pytest/_code/__init__.py", line 2, in <module>
    from .code import Code
  File "/home/heet/anaconda3/envs/mdanalysis-dev/lib/python3.10/site-packages/_pytest/_code/code.py", line 59, in <module>
    from exceptiongroup import BaseExceptionGroup
ModuleNotFoundError: No module named 'exceptiongroup'
```

**Reply by IAlibay** [reply]
How odd, what happens if you run `python -Ic "import MDAnalysis; print(MDAnalysis.__version__)"`?

**Reply by HeetVekariya** [reply]
Output:
```bash
(mdanalysis-dev) heet@heet:~/Work/Development/mdanalysis/testsuite/MDAnalysisTests$ python -Ic "import MDAnalysis; print(MDAnalysis.__version__)"
/home/heet/anaconda3/envs/mdanalysis-dev/lib/python3.10/site-packages/h5py/__init__.py:36: UserWarning: h5py is running against HDF5 1.14.3 when it was built against 1.14.2, this may cause problems
  _warn(("h5py is running against HDF5 {0} when it was built against {1}, "
2.6.1
```

**Reply by HeetVekariya** [reply]
I have also run this command: ```pip show MDAnalysisTests```
Output:
```bash
(mdanalysis-dev) heet@heet:~/Work/Development/mdanalysis/testsuite/MDAnalysisTests$ pip show MDAnalysisTests
Name: MDAnalysisTests
Version: 2.6.1
Summary: MDAnalysis testsuite
Home-page: https://www.mdanalysis.org
Author: MDAnalysis Development Team
Author-email: mdanalysis@numfocus.org
License: GPL-3.0-or-later
Location: /home/heet/anaconda3/envs/mdanalysis-dev/lib/python3.10/site-packages
Requires: hypothesis, MDAnalysis, pytest
Required-by: 
```

**Reply by IAlibay** [reply]
@HeetVekariya The 2.6.1 indicates that you aren't running a development build but instead the last released version of MDAnalysis.

How did you install the development version of MDAnalysis?

**Reply by HeetVekariya** [reply]
- To install MDAnalysis for development, i have used this command: ```conda create --name mdanalysis-dev "python=3.10"```, as ```conda create --name mdanalysis-dev "python>=3.9"``` gives dependencies error.

**Reply by IAlibay** [reply]
@HeetVekariya did you follow the rest of the steps? i.e. "Building MDAnalysis"?

**Reply by HeetVekariya** [reply]
Then i run these commands to set up the dev environment.
```bash
1. conda activate mdanalysis-dev
2. conda install -c conda-forge \
  'Cython>=0.28' \
  'numpy>=1.21.0' \
  'biopython>=1.80' \
  'networkx>=2.0' \
  'GridDataFormats>=0.4.0' \
  'mmtf-python>=1.0.0' \
  'joblib>=0.12' \
  'scipy>=1.5.0' \
  'matplotlib>=1.5.1' \
  'tqdm>=4.43.0' \
  'threadpoolctl'\
  'packaging' \
  'fasteners' \
  'netCDF4>=1.0' \
  'h5py>=2.10' \
  'chemfiles>=0.10' \
  'pyedr>=0.7.0' \
  'pytng>=0.2.3' \
  'gsd>3.0.0' \
  'rdkit>=2020.03.1' \
  'parmed' \
  'seaborn' \
  'scikit-learn' \
  'tidynamics>=1.0.0'

# documentation dependencies
conda install -c conda-forge sphinx pybtex pybtex-docutils \
    sphinxcontrib-bibtex sphinx_rtd_theme sphinx-sitemap
python -m pip install msmb_theme==1.2.0

3. # go to the mdanalysis source directory
cd mdanalysis/

# Build and install the MDAnalysis package
cd package/
python -m pip install -e .

# Build and install the test suite
cd ../testsuite/
python -m pip install -e .
```

**Reply by IAlibay** [reply]
@HeetVekariya - Apologies, did you mean that you have already run those steps or are you asking if you should. If it is the latter, then definitely please do so.

**Reply by HeetVekariya** [reply]
Yes, i have already run that steps.

**Reply by IAlibay** [reply]
This should return you an environment where the MDAnalysis version is 2.7.0-dev0. I'm not sure why you are getting 2.6.1.

Did you install MDAnalysis on top of the environment by any chance?

**Reply by HeetVekariya** [reply]
I don't think so, that i have installed MDAnalysis.

**Reply by IAlibay** [reply]
I'm not sure that I have a better solution for you here than to try to do this again in a brand new environment (or delete the current environment and recreate it).

**Reply by HeetVekariya** [reply]
I will do it after merging of my open PR's. Thank you for guiding me.

**Reply by HeetVekariya** [reply]
Also i have seen this [Travis](https://userguide.mdanalysis.org/stable/testing.html#continuous-integration-tools) PR check on commits in MDAnalysis, if it action doesn't exists, we should remove it from docs along the image associated with it, right ?

**Reply by RMeli** [reply]
One possibility to not re-do the whole environment would be to try to uninstall MDAnalysis with both `conda` and `pip` and then re-install it with `python -m pip install -e .`.

> Also i have seen this [Travis](https://userguide.mdanalysis.org/stable/testing.html#continuous-integration-tools) PR check on commits in MDAnalysis, if it action doesn't exists, we should remove it from docs along the image associated with it, right?

We used to have Travis instead of GitHub Actions. What you are looking at is the User Guide for version `2.6.1` which still has this information. If you look at the `2.7.0-dev0` or `dev` versions of the User Guide, you will see that Travis is not there and there is updated information (about [GitHub Actions](https://userguide.mdanalysis.org/dev/testing.html#github-actions), for example). It has been fixed in https://github.com/MDAnalysis/UserGuide/pull/291.

**Reply by HeetVekariya** [reply]
Okay, can you please let me know how can i access the ```2.7.0-dev0``` or ```dev``` docs, such that i only raise relevant queries only :smile:

**Reply by IAlibay** [reply]
@HeetVekariya the development documentation is found under:

https://docs.mdanalysis.org/dev/

and 

https://userguide.mdanalysis.org/dev/

**Reply by RMeli** [reply]
You can also toggle the version on the webpage, at the bottom of the menu on the left.
